tee-worker/litentry/primitives/src/lib.rs

```diff
@@ -157,26 +148,18 @@ impl LitentryMultiSignature {
 			Self::Ethereum(ref sig) =>
 				return verify_evm_signature(evm_eip191_wrap(msg).as_slice(), sig, signer)
 					|| verify_evm_signature(msg, sig, signer),
-			Self::EthereumPrettified(ref sig) => {
-				let prettified_msg =
-					LITENTRY_PRETTIFIED_MESSAGE_PREFIX.to_string() + &hex_encode(msg);
-				let msg = prettified_msg.as_bytes();
-				return verify_evm_signature(evm_eip191_wrap(msg).as_slice(), sig, signer)
-					|| verify_evm_signature(msg, sig, signer)
-			},
 			_ => false,
 		}
 	}
 
 	fn verify_bitcoin(&self, msg: &[u8], signer: &Address33) -> bool {
 		match self {
 			Self::Bitcoin(ref sig) =>
-				verify_bitcoin_signature(hex::encode(msg).as_str(), sig, signer),
-			Self::BitcoinPrettified(ref sig) => {
-				let prettified_msg =
-					LITENTRY_PRETTIFIED_MESSAGE_PREFIX.to_string() + &hex_encode(msg);
-				verify_bitcoin_signature(prettified_msg.as_str(), sig, signer)
-			},
+				verify_bitcoin_signature(hex::encode(msg).as_str(), sig, signer)
+					|| match std::str::from_utf8(msg) {
+						Err(_) => false,
+						Ok(prettified) => verify_bitcoin_signature(prettified, sig, signer),
```

Comment:
Hmm why do we verify the utf8 string? It seems to be different from what we did.

Reply:
For the prettified version. 

Maybe there is another way but this is what I found.

`TrustedCall` controls the signature prefix based on the call and creates the raw + prettified versions:

https://github.com/litentry/litentry-parachain/blob/31050aeb44e159cba5128b3d59886842f68b168c/tee-worker/app-libs/stf/src/trusted_call.rs#L357-L360

I noticed that for Bitcoin signature to work, the prettified version should be encoded as utf8. It won't work when `&str` is passed as `&u8`, so I had to decode back to utf8. I tried to mimic what we currently have:

raw check as hex:
https://github.com/litentry/litentry-parachain/blob/534c2d369eaec2a8a93c062a5a5e8a6f7cd5d7cd/tee-worker/litentry/primitives/src/lib.rs#L174

str prettified check:
https://github.com/litentry/litentry-parachain/blob/534c2d369eaec2a8a93c062a5a5e8a6f7cd5d7cd/tee-worker/litentry/primitives/src/lib.rs#L178

For `verify_bitcoin`, it'd be more convenient to expect `msg` as `&str` but I didn't want to change it because it would involve reworking any other `verify_*` function.

Reply:
I aligned with Kai about the utf_8 conversion. By looking at the code we thought it wouldn't work, but as shown in the video from the description it works well for Bitcoin, which I have tested throughly.

We agreed the issue comes mainly from `verify_bitcoin_signature` which expects a `str` instead of raw bytes.